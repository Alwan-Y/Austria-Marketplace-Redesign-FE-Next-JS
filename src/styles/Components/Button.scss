.btn {
  font-size: 0.875rem;

  &__black {
    background-color: $black;

    &:hover {
      opacity: 0.8;
    }
  }

  &__default {
    &:hover {
      color: $primary;
    }
  }
}