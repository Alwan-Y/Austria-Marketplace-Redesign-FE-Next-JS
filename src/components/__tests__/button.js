import Button from '../commons/Button';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';

test('Button renders correctly', () => {
  const { container } = render(<Button type="primary">Submit</Button>);

  expect(container).toMatchSnapshot();
});

test('Button primary has correct classnames', () => {
  const { container } = render(<Button type="primary">Submit</Button>);

  expect(container.firstChild).toHaveClass('btn');
  expect(container.firstChild).toHaveClass('btn-primary');
});

test('Trigger onClick while click the button', () => {
  const onClick = jest.fn();
  render(
    <Button type="primary" onClick={onClick}>
      Submit
    </Button>
  );

  userEvent.click(screen.getByText('Submit'));
  expect(onClick).toHaveBeenCalledTimes(1);
});

test('Button Outline', () => {
  const { container } = render(
    <Button className="btn btn-outline-primary">Submit</Button>
  );

  expect(container.firstChild).toHaveClass('btn-outline-primary');
});

test('Button size', () => {
  const {container} = render(<Button className="btn btn-lg">Submit</Button>)

  expect(container.firstChild).toHaveClass('btn btn-lg')
})

test('Button Block', () => {
  const {container} = render(<Button className="btn btn-block">Submit</Button>)

  expect(container.firstChild).toHaveClass('btn btn-block')
})


