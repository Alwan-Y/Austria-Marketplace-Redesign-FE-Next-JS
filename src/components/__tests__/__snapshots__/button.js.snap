// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<div>
  <button
    class="button btn btn-primary"
    type="button"
  >
    Submit
  </button>
</div>
`;
